components/codec/src/bits/buffered_read.rs

```diff
@@ -0,0 +1,224 @@
+// Copyright 2023 The CeresDB Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::bits::{Bit, Error};
+
+/// BufferedReader
+///
+/// BufferedReader encapsulates a buffer of bytes which can be read from.
+#[derive(Debug)]
+pub struct BufferedReader<'a> {
+    bytes: &'a [u8], // internal buffer of bytes
+    index: usize,    // index into bytes
+    pos: u32,        // position in the byte we are currently reading
```

Comment:
Put the comments above every field. And I suggest renaming the two fields `index` and `pos` to `byte_idx` and `bit_idx`.